game-core/src/main/java/games/strategy/engine/data/unit/ability/ConvertUnitAbility.java

```diff
@@ -0,0 +1,64 @@
+package games.strategy.engine.data.unit.ability;
+
+import games.strategy.engine.data.UnitType;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+
+/** Unit Ability to convert CombatUnitAbilities into other CombatUnitAbilities or into nothing */
+@Data
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Builder(toBuilder = true)
+public class ConvertUnitAbility {
+
+  enum Faction {
+    ALLIED,
+    ENEMY
+  }
+
+  @NonNull String name;
+  /** The unit types that have this ability */
+  @NonFinal Collection<UnitType> attachedUnitTypes;
+
+  /** Does this affect enemy and/or allied units */
+  @Builder.Default Collection<Faction> factions = List.of(Faction.ENEMY);
+
+  /** The unitAbility that will be converted */
+  @NonNull CombatUnitAbility from;
+
+  /** If to is null, then this just removes the "from" UnitAbility */
+  CombatUnitAbility to;
+
+  /**
+   * Can the attachedTo from "other" be combined with this?
+   *
+   * <p>This is only used to merge auto created ConvertUnitAbilities.
+   *
+   * <p>The name and attachedTo is not important in deciding whether the attachedTo can be combined.
+   */
+  public boolean canMergeAttachedUnitTypes(final ConvertUnitAbility other) {
+    // build a new ConvertUnitAbility based off of other but with this ConvertUnitAbility's name
+    // and attachedUnitTypes as they are not relevant for merging.
+    return equals(
+        other.toBuilder().name(this.name).attachedUnitTypes(this.attachedUnitTypes).build());
```

Comment:
It's not clear why we are using a builder to do an equality check. Does this mean that the two are equal if all fields match *except* the two listed here? 

If that is the case, there is a built in way to do that: `@EqualsAndHashCode(exclude = {})`

Could you elaborate on why this technique with the partial toBuilder copy is used here?

Reply:
I actually went with the `EqualsAndHashCode` code before but then I realized that I actually needed those properties to be part of the equality except when trying to merge the attachedUnitTypes.  I then created a separate equals method and built out the equal checks and realized it was identical to the equal check except for those two fields.  So creating a temp object with an override of those two fields allowed me to simplify the code.

I tried to explain all of that in the comments.  It says that the "name" and "attachedUnitTypes" are not relevant when merging attachedUnitTypes.